polkadot/runtime/parachains/src/scheduler.rs

```diff
@@ -228,10 +154,28 @@ impl<T: Config> Pallet<T> {
 	/// Called by the initializer to finalize the scheduler pallet.
 	pub(crate) fn initializer_finalize() {}
 
-	/// Called before the initializer notifies of a new session.
-	pub(crate) fn pre_new_session() {
-		Self::push_claim_queue_items_to_assignment_provider();
-		Self::push_occupied_cores_to_assignment_provider();
+	/// Called before the initializer notifies any modules of a new session.
+	pub(crate) fn pre_new_session(occupied_cores: impl Iterator<Item = (CoreIndex, ParaId)>) {
+		// The formerly occupied cores will have their candidates dropped in the inclusion module.
+		// We only need to report them as processed. These are candidates that took longer than 1
+		// block to become available and unfortunately they happened to be right before the session
+		// change. We don't push back to the assignment provider for simplicity.
+		for (core_index, para_id) in occupied_cores {
+			T::AssignmentProvider::report_processed(para_id, core_index);
+		}
+	}
+
+	/// Called after the initializer notifies all modules of a new session.
+	pub(crate) fn post_new_session(prev_core_count: u32) {
```

Comment:
> Note this call is redundant in nature, as we already have initializer_on_new_session and we also do core count handling there. It would be good to have everything in one call. (To avoid divergence of logic.)

Yes, this is a good idea. I did it this way initially because I thought it'd be important to have the coretime assigner first reconfigured with the new session change before pushing back assignments to it. but doesn't seem to matter.

> You should have all the data you need in SessionChangeNotification. Probably a good time to get rid of the pretty much obsolete AssignmentProvider::session_core_count and reading configuration directly.

This is true only if we remove the assigner_parachains pallet. I think this makes sense since all production networks moved to coretime anyway. right? should I open an issue for removing the code of this pallet?

Reply:
opened https://github.com/paritytech/polkadot-sdk/issues/5970